fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1454,6 +1457,145 @@ void partitionFieldPredicateDetectionTest(boolean isSynthetic) {
                     assertEquals(expectedLuceneComparisonQuery, luceneComparisonQuery);
                 }
             }
+
+            // test some "negative" use cases, asserting we won't mistakenly detect a partition field predicate and use
+            // it to optimize the query
+            QueryComponent textSearchPredicate = new LuceneQueryComponent(luceneSearch, List.of(textFieldName));
+            QueryComponent secondTextSearchPredicate = new LuceneQueryComponent(luceneSearch2, List.of(textFieldName));
+            QueryComponent luceneDslPredicate = new LuceneQueryComponent(luceneSearch + " " + partitionFieldName + ":[1 TO 10]", List.of(textFieldName, partitionFieldName));
+            final QueryComponent partitionFieldPredicate;
+            final QueryComponent secondPartitionFieldPredicate;
+            final QueryComponent groupPredicate;
+            if (isSynthetic) {
+                partitionFieldPredicate = Query.field("complex").matches(Query.field("timestamp").greaterThan(15L));
+                secondPartitionFieldPredicate = Query.field("complex").matches(Query.field("timestamp").greaterThan(55L));
+                groupPredicate = Query.field("complex").matches(Query.field("group").equalsParameter("group_value"));
+            } else {
+                partitionFieldPredicate = Query.field(partitionFieldName).greaterThan(15L);
+                secondPartitionFieldPredicate = Query.field(partitionFieldName).greaterThan(55L);
+                groupPredicate = Query.field("group").equalsParameter("group_value");
+            }
+            Map<QueryComponent, QueryPlanningExpectation> filterToOutcomeMap = Map.of(
+                    // full disjunction
+                    Query.or(groupPredicate, textSearchPredicate, partitionFieldPredicate), new QueryPlanningExpectation(
+                            // planning raises exception
+                            false,
+                            // planning produces non-lucene plan
+                            true,
+                            // synthetic planning raises exception
+                            true,
+                            // syntehtic planning produces non-lucene plan
+                            false,
```

Comment:
Given the comments, perhaps it would be better to have an `enum` with `NonLucenePlan` and `Exception` values, and then have the expectation take one for synthetic, and one for non-synthetic.

Reply:
It might also be ok to just silently ignore all of these. As far as this test is concerned it just cares that it doesn't ever pick the partitioning key.
If you do that, it would probably be good to have happy paths using the same query components, so something like:
```
Map<QueryComponent, Boolean> filterToOutcomeMap = Map.of(
    Query.or(groupPredicate, textSearchPredicate, partitionFieldPredicate), false,
    ....
    Query.and(groupPredicate, textSearchPredicate, partitionFieldPredicate), true))
```

And then you know that all the query construction and execution is good, and it is isolating to the fact that you are using `or`, etc.

Reply:
It makes things a bit more complex to try to jam all of these situations into a single flow; for example, some filters will cause exceptions to be thrown. Also, the "positive" cases try all different combinations of operators etc, which the negative cases don't have to. I will keep them separate, and will split into two tests as suggested in a previous comment.